### DISCUSSION TITLE
20231228 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by cgbosse** [nautilus-cyberneering/.github#75] [Meetings]
Time: 8:30 GMT+0  Canary Islands Time (9:30 in Vaduz)

Location:
<https://meeting.hodl10.io/ConsistentClassroomsMergeSideways>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x
- MVanDijke(@WarmBeer) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by mario-nt** [comment]
### GUI

- https://github.com/torrust/torrust-index-gui/issues/401
- https://github.com/torrust/torrust-index-gui/issues/278
- Research on how to implement light/dark mode for the GUI

### Index

- https://github.com/torrust/torrust-index/issues/39

### Other

- Set up new PC

**Comment by josecelano** [comment]
### Tracker

- Issue: [Enable test should_fail_with_ssl_enabled_and_bad_ssl_config](https://github.com/torrust/torrust-tracker/issues/521). **Waiting for feedback**.
- Issue: [Tracker logs are too verbose](https://github.com/torrust/torrust-tracker/issues/539). I finally keep the log for each request but with a different format.
- PR review: [chore: update cargo deps](https://github.com/torrust/torrust-tracker/pull/553)
- PR review: [Update labels.json](https://github.com/torrust/torrust-tracker/pull/555). New contributor @ShrirangB 🚀 .

### Index

- Issue: [Patch for Tracker API response is not working](https://github.com/torrust/torrust-index/issues/391)
- Issue: [Fix tracker API error token not valid](https://github.com/torrust/torrust-index/issues/419).

### Index GUI

- Issue: [authentication error: missing authentication key param for private tracker](https://github.com/torrust/torrust-index-gui/issues/412). [Waiting for feedback](https://github.com/torrust/torrust-index-gui/issues/412#issuecomment-1866890915).

### Website

- PR review: [widen text on larger screens and change GmbH to SL
](https://github.com/torrust/torrust-website/pull/123)

### Short research

Short research about decentralized indexes and how to scrape the DHT and get torrent metadata:

- https://www.tribler.org/
- https://github.com/urbanguacamole/torrent-paradise
- https://github.com/boramalper/magnetico
- https://github.com/algoprog/rivr-search

### Open Discussions

- https://github.com/torrust/torrust-tracker/discussions/545
- https://github.com/torrust/torrust-tracker/discussions/546
- https://github.com/torrust/torrust-tracker/discussions/135 or collect data with prometheus. It would be nice a [prometheus-ghrafana](https://grafana.com/docs/grafana/latest/getting-started/get-started-grafana-prometheus/) container setup for the live demo.
- https://github.com/torrust/torrust-tracker/discussions/548
- https://github.com/torrust/torrust-tracker/discussions/547
- https://github.com/torrust/torrust-index/discussions/418

**Comment by cgbosse** [comment]
## Sent email by @WarmBeer 

I have been sick for a couple of days already and have been very busy with the holidays, so there is not much I can contribute.

My plan for the following week is to implement DashMap for the torrent tracker in an attempt to make torrent repository reads and writes even faster. Then I will work on adding a max limit to the torrent repository.
